src/inspectorscripts.ts

```diff
@@ -25,29 +25,43 @@ _jupyterlab_variableinspector_nms = NamespaceMagics()
 _jupyterlab_variableinspector_Jupyter = get_ipython()
 _jupyterlab_variableinspector_nms.shell = _jupyterlab_variableinspector_Jupyter.kernel.shell
 
-try:
-    import numpy as np
-except ImportError:
-    np = None
+np = None
+pd = None
+pyspark = None
+tf = None
 
-try:
-    import pandas as pd
-except ImportError:
-    pd = None
 
-try:
-    import pyspark
-except ImportError:
-    pyspark = None
+def _check_imported():
+    global np, pd, pyspark, tf
+    if 'numpy' in sys.modules:
+        # don't really need the try
+        try:
+            import numpy as np
+        except ImportError:
+            np = None
 
-try:
-    import tensorflow as tf
-    import keras.backend as K
-except ImportError:
-    tf = None
+    if 'pandas' in sys.modules:
+        try:
+            import pandas as pd
+        except ImportError:
+            pd = None
+
+    if 'pyspark' in sys.modules:
+        try:
+            import pyspark
+        except ImportError:
+            pyspark = None
+
+    if 'tensorflow' in sys.modules or 'keras' in sys.modules:
+        try:
+            import tensorflow as tf
+            import keras.backend as K
+        except ImportError:
+            tf = None
 
 
 def _jupyterlab_variableinspector_getsizeof(x):
```

Comment:
Do we need to call `_check_imported()` for each subfunction? Shouldn't it be enough to call it once when the script is initialized?

Reply:
@lckr yes, otherwise it will continue to evaluate the module name as `None`

Reply:
@lckr you call it on each function so that once the user imports it, you update the module name to be correct e.g. before the user imports `tensorflow`, `tf=None`, but then once they do, `tf=tensorflow`

Reply:
Thanks but my point is that is should be enough to call `_check_imported()` only in the `def _jupyterlab_variableinspector_dict_list()` method which is executed each time the kernel updates and also calls all other methods.
Therefore we would save us the multiple executions of `_check_imported()` that are not really required, or do you see this differently?

Reply:
Sounds good so long as it's executed first, I wasn't sure what the code paths were